src/local/butler/reproduce.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""reproduce.py reproduces test cases locally."""
+
+from __future__ import print_function
+
+import os
+import six
+
+from src.python.fuzzing import tests
+from src.python.system import environment
+
+
+class _SimplifiedTestcase(object):
+  """Minimal representation of a test case."""
+
+  def __init__(self, testcase_json):
+    self.crash_state = testcase_json['crash_state']
+    self.security_flag = testcase_json['security_flag']
+    self.gestures = testcase_json['gestures']
+    self.flaky_stack = testcase_json['flaky_stack']
+
+    # Custom field not included in real test cases. Used in environment setup.
+    self.job_definition = testcase_json['job_definition']
+
+
+def _get_testcase(_):
+  """Retrieve the json representation of the test case with the given id."""
+  # TODO(mbarbella): Actually fetch the test case info from the server.
+  testcase_json = {
+      'crash_state': '',
+      'security_flag': False,
+      'gestures': [],
+      'flaky_stack': False,
+      'job_definition': 'APP_NAME = echo\nAPP_ARGS = -n\n',
+  }
+
+  return _SimplifiedTestcase(testcase_json)
+
+
+def _download_testcase(_):
+  """Download the test case and return its path."""
+  # TODO(mbarbella): Implement this.
+  return '/tmp/blah'
+
+
+def _prepare_environment(testcase, build_directory):
+  """Prepare environment variables based on the test case and build path."""
+  environment.set_default_vars()
+
+  # Environment variables used by the tool.
+  environment.set_value('APP_DIR', build_directory)
+  environment.set_value('BUILDS_DIR', build_directory)
+  environment.set_value('INPUT_DIR', '/tmp/fixme')
```

Comment:
We don't need all these vars like BOT_TMPDIR, INPUT_DIR, etc to be set. Please just call CF func
environment.set_bot_environment() and remove environment.set_default_vars() [that is called as well by set_bot_environment].

Reply:
Done